public/src/components/switchboard.tsx

```diff
@@ -1,456 +1,211 @@
-      this.previousStateFromServer = serverData;
-      this.setState({
-        ...serverData.value,
-      });
-    });
-  }
-
-  updateOneOffPaymentMethodSwitch(paymentMethod: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        oneOffPaymentMethods: {
-          [paymentMethod]: { $set: switchState },
-        },
-      }),
-    );
-  }
-
-  updateRecurringPaymentMethodSwitch(paymentMethod: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        recurringPaymentMethods: {
-          [paymentMethod]: { $set: switchState },
-        },
-      }),
-    );
-  }
-
-  updateFeatureSwitch(switchName: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        experiments: {
-          [switchName]: {
-            state: { $set: switchState },
-          },
-        },
-      }),
+    updatedState[groupId].switches[mySwitchId].state = event.target.checked
+      ? SwitchState.On
+      : SwitchState.Off;
+
+    setData(updatedState);
+    setNeedToSaveDataWarning(true);
+  };
+
+  const createSwitchesFromGroupData = (
+    mySwitch: [string, Switch],
+    groupId: string,
+  ): JSX.Element => {
+    const [switchId, switchData] = mySwitch;
+
+    return (
+      <FormControlLabel
+        label={switchData.description}
+        checked={switchData.state === 'On' ? true : false}
+        control={<SwitchUI />}
+        onChange={(event): void => updateSwitchSetting(groupId, mySwitch, event)}
+        value={switchId}
+        key={switchId}
+      />
     );
-  }
-
-  saveSwitches = (): void => {
-    const newState = update(this.previousStateFromServer, {
-      value: { $set: this.state },
-    });
-
-    saveSupportFrontendSettings(SupportFrontendSettingsType.switches, newState)
-      .then(resp => {
-        if (!resp.ok) {
-          resp.text().then(msg => alert(msg));
-        }
-        this.fetchStateFromServer();
-      })
-      .catch(resp => {
-        alert(`Error while saving: ${resp}`);
-        this.fetchStateFromServer();
-      });
   };
 
-  render(): React.ReactNode {
-    const { classes } = this.props;
+  const createGroupsFromData = (group: [string, SwitchGroup]): JSX.Element => {
+    const [groupId, groupData] = group;
 
     return (
+      <FormControl className={classes.formControl} key={groupId}>
+        <FormLabel>{groupData.description}</FormLabel>
+        {Object.entries(groupData.switches).map(mySwitch =>
+          createSwitchesFromGroupData(mySwitch, groupId),
+        )}
+      </FormControl>
+    );
+  };
+
+  const createSwitchFields = (): JSX.Element => {
+    return <>{Object.entries(data).map(group => createGroupsFromData(group))}</>;
+  };
+
+  // This isn't working because original state doesn't get conserved in this component
+  // - Need to consider whether we need a 'reset' button?
```

Comment:
We'd need to add a `resetData` callback to `withS3Data` to enable this, to re-fetch from S3.
Perhaps it's not necessary - the user can just refresh the page?

Reply:
I've taken out the refresh functionality - like you say, there's no need for it as user can just refresh page or navigate away and return